### PR TITLE
CSS for all diagram items

### PR DESCRIPTION
**Pull request by amolenaar**
### PR Type
What kind of change does this PR introduce?

- [ ] Bug fix
- [x] Feature
- [ ] Chore (refactoring, formatting, local variables, other cleanup)
- [ ] Documentation content changes

### What is the current behavior?

Issue Number: #3023

### What is the new behavior?

New CSS capabilities:

* All inline styling has been removed.
* Inline styling is no longer supported by shapes.
* icons can be styled with `icon` nodes.
* compartments can be styles as `compartment` nodes.
* Add `:first-child` selector.
* Use the default `stereotype_watches()` function where possible
* Do not propagate padding to child nodes. That can cause padding to be applied multiple times.

### Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### Other information

* It may be that properties like `text-align` and `font-*` leak through to child items in a diagram.
* The only part of a diagram not styled is the diagram type/name pentagon on the top-left.

### CODE REVIEW

**Review by danyeaw — APPROVED** [review]
Thanks @amolenaar, this looks great!

**Review by danyeaw — APPROVED** [review]

File: gaphor/UML/actions/tests/test_objectnode.py [hunk 1]
```diff
@@ -13,7 +13,7 @@ def test_object_node(diagram, element_factory):
 
 def test_name(create):
     node = create(ObjectNodeItem, UML.ObjectNode)
-    name = node.shape.icon.children[1]
+    name = node.shape.icon.chid.children[1]
```
[the remark this answers is not in the document]
Comment (danyeaw) [reply gaphor/UML/actions/tests/test_objectnode.py:16]: I think it is a typo, chid instead of child.

[the remark this answers is not in the document]
Comment (amolenaar) [reply gaphor/UML/actions/tests/test_objectnode.py:16]: Argh!

[the remark this answers is not in the document]
Comment (amolenaar) [reply gaphor/UML/actions/tests/test_objectnode.py:16]:
```suggestion
    name = node.shape.icon.child.children[1]
```

File: gaphor/UML/actions/objectnode.py [hunk 2]
```diff
@@ -33,10 +33,7 @@ def __init__(self, diagram, id=None):
...
                 Box(
                     text_stereotypes(self),
-                    Text(text=lambda: self.subject.name or ""),
-                    style={
-                        "padding": (4, 12, 4, 12),
-                    },
+                    text_name(self),
                     draw=draw_border,
                 ),
                 Text(
```
Comment (amolenaar) [inline gaphor/UML/actions/objectnode.py:39]: `upperBound` and `ordering` are not tagged by CSS nodes.

File: gaphor/diagram/shapes.py [hunk 3]
```diff
@@ -177,6 +176,14 @@ def draw_diamond(
...
 
 
+def draw_all(*draw_functions):
+    def _draw(box, context, bounding_box):
+        for draw in draw_functions:
+            draw(box, context, bounding_box)
+
+    return _draw
+
+
```
Comment (amolenaar) [inline gaphor/diagram/shapes.py:186]: Do we need this still?

**Comment by amolenaar** [comment]
@danyeaw I added a lot more changes to this PR.

**Review by danyeaw — APPROVED** [review]
Thanks @amolenaar!! This is such a nice improvement, it will be much easier to create new toolboxes and maintain the ones we have with all the style centralized like this! :star_struck:

**Comment by amolenaar** [comment]
A next step may be to create some style sheets, so users can pick from a few predefined styles.

**Pull request merged**